docs/pages/machine-id/troubleshooting.mdx

```diff
@@ -0,0 +1,286 @@
+will not be able to connect.
+
+When a bot's identity expires, certain server-side parameters like the
+certificate generation counter must be reset. The simplest way to accomplish
+this is by removing and recreating the bot, which purges all server-side data
+and issues a new joining token.
+
+### Resolution
+
+Remove and recreate the bot, replacing the name and role list as desired:
+
+```code
+$ sudo tctl bots rm example
+$ sudo tctl bots add example --roles=access
+```
+
+Copy the resulting join token into the existing bot config - either the
+`--token` CLI flag or the `onboarding.token` parameter in `tbot.yaml` - and
+restart the bot. It will detect the new token and rejoin the cluster as normal.
+
+## SSH connections fail with `ssh: handshake failed: ssh: unable to authenticate`
+
+### Symptoms
+
+When attempting to connect to a node via SSH, connections fail with an error
+like the following:
+
+```code
+$ ssh -F /opt/machine-id/ssh_config bob@node.example.com                                                                                                                           timothyb89/v10/machine-id-app-access 
+ERROR: ssh: handshake failed: ssh: unable to authenticate, attempted methods [none publickey], no supported methods remain
+
+ERROR: unable to execute tsh
+executing `tsh proxy`
+exit status 1
+
+kex_exchange_identification: Connection closed by remote host
+Connection closed by UNKNOWN port 65535
+```
+
+In particular, note the `ssh: unable to authenticate` message.
+
+### Explanation
+
+This can occur when attempting to log into the node as a user not listed as a
+principal on the SSH certificate.
+
+This can be verified by viewing the `tbot` logs and looking for the log message
+when impersonated certificates for the matching destination were renewed.
+
+In the following example, the only principal listed for the identity in
+`/opt/machine-id` is `alice` (via the `access` role):
+```text
+INFO [TBOT]      Successfully renewed impersonated certificates for directory /opt/machine-id, valid: after=2022-07-21T21:49:26Z, before=2022-07-21T22:50:26Z, duration=1h1m0s | kind=tls, renewable=false, disallow-reissue=true, roles=[access], principals=[alice -teleport-internal-join], generation=0 tbot/renew.go:630
+```
+
+... however, the SSH command attempted to login as `bob`.
+
+### Resolution
+
+Ensure the bot identity is allowed to login as the requested user by taking any
+of the following actions:
+
+ 1. Changing the SSH command to login as an allowed user
+ 2. Modifying the `access` role to allow the `alice` principal
+ 3. Adding an additional role granting login via the `bob` principal
+
+Note that if roles are added or modified, the certificates will need to be
+renewed for the changes to take effect. The bot will renew certificates on its
+own after the renewal interval (by default, 20 minutes), but you can trigger a
+renewal immediately by either restarting the `tbot` process or sending it a
```

Comment:
What does "restarting the `tbot` process" mean here? `systemctl restart tbot`?

Reply:
I've added an additional `systemctl` example to the code block to hopefully clarify this.